src/state/slices/opportunitiesSlice/resolvers/thorchainsavers/index.ts

```diff
@@ -195,38 +198,47 @@ export const thorchainSaversStakingOpportunitiesMetadataResolver = async ({
       ) as AssetId[]
 
     const totalRuneAmount = reservePositions.pools.reduce((acc, pool) => {
-      const share = bnOrZero(pool.liquidityUnits).div(poolsByAsset[pool.pool].pool_units)
-      const runeAmount = share.times(poolsByAsset[pool.pool].balance_rune)
+      const assetId = poolAssetIdToAssetId(pool.pool)
+
+      if (!assetId) return acc
+
+      const share = bnOrZero(pool.liquidityUnits).div(poolsByAssetid[assetId].pool_units)
+      const runeAmount = share.times(poolsByAssetid[assetId].balance_rune)
 
       return acc.plus(runeAmount)
     }, bn(0))
 
     const { underlyingAssetRatiosBaseUnit, underlyingAssetWeightPercentageDecimal } =
       reservePositions.pools.reduce(
-        (acc, pool, i) => {
-          const share = bnOrZero(pool.liquidityUnits).div(poolsByAsset[pool.pool].pool_units)
+        (acc, pool) => {
+          const assetId = poolAssetIdToAssetId(pool.pool)
+
+          if (!assetId) return acc
```

Comment:
ditto on `poolsByAssetid[assetId]` and assign to a constant

Reply:
In [6ff23f4](https://github.com/shapeshift/web/pull/7459/commits/6ff23f48c0919cc936619a1689ec044d887f65d8)